src/core/protocol-authorization.ts

```diff
@@ -18,25 +17,22 @@ const methodToAllowedActionMap: Record<string, ProtocolAction> = {
 export class ProtocolAuthorization {
 
   /**
-   * Performs protocol-based authorization against the given message.
-   * @param recordsWrite Either the incomingMessage itself if the incoming is a RecordsWrite,
-   *                     or the latest RecordsWrite associated with the recordId being read.
-   * @throws {Error} if authorization fails.
+   * Performs validation on the structure of RecordsWrite messages that use a protocol.
+   * @throws {Error} if validation fails.
    */
-  public static async authorize(
+  public static async validate(
```

Comment:
Can we rename to more precisely show the intent? I feel we have started to use "validate structure" in multiple context, e.g. there are "structure validations" that happens in `parse()` which begs the question why this couldn't be done there for people not familiar. Names below may make good alternate candidates:

`validateReferentialIntegrity()`, `validateReferentialConsistency()`

Reply:
Agreed on renaming -- I like `validateReferentialIntegrity`. I considered putting this inside `RecordsWrite.parse()` but decided against it because this method requires `tenant` and `messageStore`, which feel outside of the scope of what `parse()` should do/have.